### PR TITLE
Utility function to assess conditional independence

### PR DESCRIPTION
**Pull request by bkayalibay**
This is an attempt at addressing issue #290. The utility function `is_independent` is located at `edward/util/random_variables.py` and I've added some tests at `tests/util/test_is_independent`.

### CODE REVIEW

File: edward/util/random_variables.py [hunk 1]
```diff
@@ -714,6 +715,93 @@ def get_variables(x, collection=None):
...
+       Query node(s).
+
+    condition : RandomVariable or list of RandomVariable, optional
+       Random variable(s) to condition on.
+
+  Returns
+    bool
+    True if a is independent of b given the random variables in condition.
+
+  References
```
Comment (dustinvtran) [inline edward/util/random_variables.py]: Can you add a bibtex entry to `docs/tex/bib.bib`? Also note the bib file is sorted chronologically by year, then sorted alphabetically by author's last name within a year.

File: edward/util/random_variables.py [hunk 2]
```diff
@@ -714,6 +715,93 @@ def get_variables(x, collection=None):
   return list(output)
 
 
+def is_independent(a, b, condition=None):
+  """Assess whether a is independent of b given the random variables in
+  condition.
+
+  Implemented using the Bayes-Ball algorithm[1].
+
+  Args
```
Comment (dustinvtran) [inline edward/util/random_variables.py]: Fields like `Args` and `Returns` require a colon at the end, c.f., other function docstrings in this file.

File: edward/util/random_variables.py [hunk 3]
```diff
@@ -714,6 +715,93 @@ def get_variables(x, collection=None):
...
+    a : RandomVariable or list of RandomVariable
```
Comment (dustinvtran) [inline edward/util/random_variables.py]: Args are written as, e.g., `a: RandomVariable or list of RandomVariable.` Note the removed space and added period. When in doubt, you can also double check the generated API docstrings by checking out the README in `docs/`, or consult with the other function docstrings.

**Comment by bkayalibay** [comment]
Thank you very much for the feedback. I've made changes to the docstring and added a bibtex entry. The function actually uses Alg 2 from Schachter (1998). If you can point me towards a more efficient approach I'd be happy to modify the implementation.

**Comment by dustinvtran** [comment]
Perfect! LGTM.

**Pull request merged**